### DISCUSSION TITLE
Alias and Type Hint Challenges in SQLModel and Pydantic Integration

### DISCUSSION THREAD

**Discussion by AlirezaHaghi** [fastapi/sqlmodel#962] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel
from pydantic import Field

class Model(SQLModel):
    id_: int = Field(alias='id')
    
model = Model(id = 23) # pylance: No parameter named "id"

print(model)
```

### Description

![image](https://github.com/tiangolo/sqlmodel/assets/89215491/b6895d2f-3e10-48c1-8c73-2dc58df44b7e)
it works correctly but there is some issue with typing in sqlmodel.
I googled on how to use PYDANTIC ALIAS NOT SQLALCHEMY ALIAS in sqlmodel and found nothing helpful.

### Operating System

Windows

### SQLModel Version

0.0.18

### Python Version

3.12.3

**Comment by import-antigravity** [comment]
you would want to use `sqlmodel.Field`, not `pydantic.Field`, right?

**Reply by YuriiMotov** [reply]
Using `sqlmodel.Field` fixes type hints, but gives `ValidationError` in run time

**Comment by ixycoexzckwpmlcu** [comment]
Maybe related: https://docs.pydantic.dev/latest/concepts/fields/#field-aliases (under "VSCode and Pyright users") and https://github.com/pydantic/pydantic/issues/5893

**Comment by YuriiMotov** [comment]
There is already open issue to track this: https://github.com/fastapi/sqlmodel/issues/203

The workaround is to add `schema_extra={"validation_alias": "id"}`:

```py
from sqlmodel import SQLModel, Field

class Model(SQLModel):
    id_: int = Field(alias='id', schema_extra={"validation_alias": "id"})

model = Model(id = 23)

print(model)
```
